app/src/common/shared/org/mozilla/vrbrowser/search/GeolocationTask.java

```diff
@@ -0,0 +1,195 @@
+package org.mozilla.vrbrowser.search;
+
+import android.os.AsyncTask;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import javax.net.ssl.HttpsURLConnection;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+
+public class GeolocationTask extends AsyncTask<Void, Void, String> {
+
+    private static final String LOGTAG = "VRB";
+
+    private static final int MAX_RETRIES = 2;
+
+    public static class GeolocationData {
+
+        private String mCountryCode;
+        private String mCountryName;
+
+        private GeolocationData(String aCountryCode, String aCountryName) {
+            mCountryCode = aCountryCode;
+            mCountryName = aCountryName;
+        }
+
+        @NonNull
+        public static GeolocationData create(@NonNull String aCountryCode, @NonNull String aCountryName) {
+            return new GeolocationData(aCountryCode, aCountryName);
+        }
+
+        public static GeolocationData parse(String aGeolocationJson) {
+            GeolocationData data = new GeolocationData(null, null);
+
+            JSONObject json;
+            try {
+                json = new JSONObject(aGeolocationJson);
+                data.mCountryCode = (String) json.get("country_code");
+                data.mCountryName = (String) json.get("country_name");
```

Comment:
we should probably handle `json.get("fallback")` if it exists too (per the [API code](https://github.com/mozilla/ichnaea/blob/36e395a/ichnaea/api/locate/views.py#L123-L124))

Reply:
per my reply elsewhere, ignore this